### PR TITLE
Highlight annotations on mobile

### PR DESCRIPTION
**Pull request by JustinHoldstock**

### CODE REVIEW

File: src/lib/annotations/MobileAnnotator.scss [hunk 1]
```diff
@@ -1,4 +1,11 @@
 $tablet: "(min-width: 768px)";
+$three-c: #ccc;
+
+@mixin border-top-bottom {
```
Comment (pramodsum) [inline src/lib/annotations/MobileAnnotator.scss]: 👍

File: src/lib/annotations/doc/DocAnnotator.js [hunk 2]
```diff
@@ -116,8 +117,10 @@ class DocAnnotator extends Annotator {
         this.highlightCurrentSelection = this.highlightCurrentSelection.bind(this);
         this.createHighlightThread = this.createHighlightThread.bind(this);
         this.createPlainHighlight = this.createPlainHighlight.bind(this);
+        this.highlightCreateHandler = this.highlightCreateHandler.bind(this);
+        this.onTouchStart = this.onTouchStart.bind(this);
 
-        this.createHighlightDialog = new CreateHighlightDialog();
+        this.createHighlightDialog = new CreateHighlightDialog(undefined, this.isMobile);
```
Comment (pramodsum) [inline src/lib/annotations/doc/DocAnnotator.js]: Is there ever a time when we create a new CreateHighlightDialog with a parentEl param specified?

File: src/lib/annotations/doc/DocHighlightDialog.js [hunk 3]
```diff
@@ -43,13 +42,57 @@ class DocHighlightDialog extends AnnotationDialog {
...
     }
 
+    /** @inheritdoc */
+    postAnnotation(textInput) {
+        const annotationTextEl = this.element.querySelector(constants.SELECTOR_ANNOTATION_TEXTAREA);
+        const text = textInput || annotationTextEl.value;
+        if (text.trim() === '') {
+            return;
+        }
+        // Convert from plain highlight to comment
```
Comment (pramodsum) [inline src/lib/annotations/doc/DocHighlightDialog.js]: new line before comment

File: src/lib/annotations/doc/CreateHighlightDialog.js [hunk 4]
```diff
@@ -308,6 +323,12 @@ class CreateHighlightDialog extends EventEmitter {
         const highlightDialogEl = document.createElement('div');
         highlightDialogEl.classList.add(CLASS_CREATE_DIALOG);
         highlightDialogEl.innerHTML = CREATE_HIGHLIGHT_DIALOG_TEMPLATE;
+        // Get rid of the caret
```
Comment (pramodsum) [inline src/lib/annotations/doc/CreateHighlightDialog.js]: new line before comment

File: src/lib/annotations/doc/DocHighlightThread.js [hunk 5]
```diff
@@ -196,7 +201,7 @@ class DocHighlightThread extends AnnotationThread {
         // If mouse is in dialog, change state to hover or active-hover
         if (docAnnotatorUtil.isInDialog(event, this.dialog.element)) {
             // Keeps dialog open if comment is pending
-            if (this.state === STATES.pending_ACTIVE) {
```
Comment (pramodsum) [inline src/lib/annotations/doc/DocHighlightThread.js]: good catch

File: src/lib/annotations/AnnotationDialog.js [hunk 6]
```diff
@@ -6,14 +6,13 @@ import { CLASS_ACTIVE, CLASS_HIDDEN } from '../constants';
 import { decodeKeydown } from '../util';
 import { ICON_CLOSE, ICON_DELETE } from '../icons/icons';
 
-const CLASS_ANNOTATION_PLAIN_HIGHLIGHT = 'bp-plain-highlight';
 const CLASS_BUTTON_DELETE_COMMENT = 'delete-comment-btn';
 const CLASS_CANCEL_DELETE = 'cancel-delete-btn';
 const CLASS_CANNOT_ANNOTATE = 'cannot-annotate';
+const CLASS_COMMENT = 'annotation-comment';
```
Comment (jeremypress) [inline src/lib/annotations/AnnotationDialog.js]: should these all be in annotation constants? Or are they only used here?

File: src/lib/annotations/doc/CreateHighlightDialog.js [hunk 7]
```diff
@@ -91,17 +91,26 @@ class CreateHighlightDialog extends EventEmitter {
     };
 
     /**
+     * Whether or not we're on a mobile device.
+     *
+     * @property {boolean}
+     */
```
Comment (jeremypress) [inline src/lib/annotations/doc/CreateHighlightDialog.js]: Convert to one-liner syntax

File: src/lib/annotations/doc/DocAnnotator.js (hunk 2 above)
Comment (JustinHoldstock) [inline src/lib/annotations/doc/DocAnnotator.js]: Yes! In this case, I didn't include a parent bc it incurs ONE unnecessary transfer from one parent to another, the first time it's shown (we move it from one pageEl to another as it's used).

File: src/lib/annotations/doc/CreateHighlightDialog.js [hunk 8]
```diff
@@ -323,6 +344,7 @@ class CreateHighlightDialog extends EventEmitter {
         highlightDialogEl.addEventListener('click', this.stopPropagation);
         highlightDialogEl.addEventListener('mouseup', this.stopPropagation);
         highlightDialogEl.addEventListener('touchend', this.stopPropagation);
+        highlightDialogEl.addEventListener('touchstart', this.stopPropagation);
```
Comment (jeremypress) [inline src/lib/annotations/doc/CreateHighlightDialog.js]: What's happening on touchstart that isn't stopped by all the other stop propagation we've done?

File: src/lib/annotations/doc/DocAnnotator.js (hunk 2 above)
Comment (pramodsum) [inline src/lib/annotations/doc/DocAnnotator.js]: 👍 Just checking because it felt weird that we were explicitly passing in undefined

Comment (jeremypress) [inline src/lib/annotations/doc/DocAnnotator.js]: Is there a way we can avoid passing in undefined? Could we make it optional with a default?

File: src/lib/annotations/doc/DocAnnotator.js [hunk 9]
```diff
@@ -675,6 +688,18 @@ class DocAnnotator extends Annotator {
...
     /**
+     * Handle touch start event.
+     * 
+     * @return {void}
+     */
+    onTouchStart() {
+        if (this.highlighter) {
+            this.highlighter.removeAllHighlights();
+        }
+        this.createHighlightDialog.hide();
```
Comment (jeremypress) [inline src/lib/annotations/doc/DocAnnotator.js]: NIT: newline after the if block

File: src/lib/annotations/AnnotationDialog.js (hunk 6 above)
Comment (JustinHoldstock) [inline src/lib/annotations/AnnotationDialog.js]: Sumedha has moved these, to be constants, in another PR

File: src/lib/annotations/doc/CreateHighlightDialog.js (hunk 7 above)
Comment (JustinHoldstock) [inline src/lib/annotations/doc/CreateHighlightDialog.js]: done

File: src/lib/annotations/doc/CreateHighlightDialog.js (hunk 4 above)
Comment (JustinHoldstock) [inline src/lib/annotations/doc/CreateHighlightDialog.js]: done

File: src/lib/annotations/doc/CreateHighlightDialog.js (hunk 8 above)
Comment (JustinHoldstock) [inline src/lib/annotations/doc/CreateHighlightDialog.js]: There's some logic in DocAnnotator that runs through and hides this, on touchstart

File: src/lib/annotations/doc/DocAnnotator.js (hunk 2 above)
Comment (JustinHoldstock) [inline src/lib/annotations/doc/DocAnnotator.js]: I can just give it a parent. The overhead of moving this node once shouldn't be too much.

File: src/lib/annotations/doc/DocAnnotator.js (hunk 9 above)
Comment (JustinHoldstock) [inline src/lib/annotations/doc/DocAnnotator.js]: done

File: src/lib/annotations/doc/DocHighlightDialog.js (hunk 3 above)
Comment (JustinHoldstock) [inline src/lib/annotations/doc/DocHighlightDialog.js]: done

**Pull request closed without merging**